core/src/main/java/org/infinispan/xsite/XSiteAdminOperations.java

```diff
@@ -403,32 +336,24 @@ private String takeOffline(String site, Integer afterFailures, Long minTimeToWai
       // TODO [pruivo] use mutable attributes to update the take offline settings?
       // https://issues.redhat.com/browse/ISPN-16780
       CacheRpcCommand command = commandsFactory.buildXSiteAmendOfflineStatusCommand(site, afterFailures, minTimeToWait);
-      XSiteResponse<Void> response = invokeOnAll(command, new VoidResponseCollector(clusterSize()));
+      XSiteResponse response = invokeOnAll(command, new VoidResponseCollector(clusterSize()));
```

Comment:
It is not a big deal, but we can change this block to a try-catch and use the `org.infinispan.remoting.transport.impl.VoidResponseCollector.validOnly()`. It will need some updates on the `invokeOnAll` though.

Reply:
You have to explain your idea better. `org.infinispan.remoting.transport.impl.VoidResponseCollector.validOnly()` record exceptions; here I'm recording the failed destinations.

Reply:
I thought something like:

```diff
diff --git a/core/src/main/java/org/infinispan/xsite/XSiteAdminOperations.java b/core/src/main/java/org/infinispan/xsite/XSiteAdminOperations.java
index 078626e4a3b..49398fdde77 100644
--- a/core/src/main/java/org/infinispan/xsite/XSiteAdminOperations.java
+++ b/core/src/main/java/org/infinispan/xsite/XSiteAdminOperations.java
@@ -28,12 +28,11 @@
 import org.infinispan.jmx.annotations.ManagedOperation;
 import org.infinispan.jmx.annotations.Parameter;
 import org.infinispan.metrics.impl.CustomMetricsSupplier;
-import org.infinispan.remoting.responses.ValidResponse;
 import org.infinispan.remoting.rpc.RpcManager;
 import org.infinispan.remoting.rpc.RpcOptions;
 import org.infinispan.remoting.transport.Address;
 import org.infinispan.remoting.transport.ResponseCollector;
-import org.infinispan.remoting.transport.ValidResponseCollector;
+import org.infinispan.remoting.transport.impl.VoidResponseCollector;
 import org.infinispan.security.AuthorizationPermission;
 import org.infinispan.security.impl.Authorizer;
 import org.infinispan.util.logging.Log;
@@ -336,29 +335,25 @@ private String takeOffline(String site, Integer afterFailures, Long minTimeToWai
       // TODO [pruivo] use mutable attributes to update the take offline settings?
       // https://issues.redhat.com/browse/ISPN-16780
       CacheRpcCommand command = commandsFactory.buildXSiteAmendOfflineStatusCommand(site, afterFailures, minTimeToWait);
-      XSiteResponse response = invokeOnAll(command, new VoidResponseCollector(clusterSize()));
-
-      //also amend locally
-      takeOfflineManager.amendConfiguration(site, afterFailures, minTimeToWait);
-
-      if (response.errors.isEmpty()) {
+      try {
+         invokeOnAll(command, VoidResponseCollector.validOnly());
          return SUCCESS;
+      } catch (Throwable t) {
+         return String.format("Could not amend for nodes: %s", t);
+      } finally {
+         //also amend locally
+         takeOfflineManager.amendConfiguration(site, afterFailures, minTimeToWait);
       }
-      return String.format("Could not amend for nodes: %s", response.errors);
    }
 
    private String incorrectSiteName(String site) {
       return "Incorrect site name: " + site;
    }
 
-   private XSiteResponse invokeOnAll(CacheRpcCommand command, ResponseCollector<XSiteResponse> responseCollector) {
+   private void invokeOnAll(CacheRpcCommand command, ResponseCollector<Void> responseCollector) {
       RpcOptions rpcOptions = rpcManager.getSyncRpcOptions();
-      CompletionStage<XSiteResponse> rsp = rpcManager.invokeCommandOnAll(command, responseCollector, rpcOptions);
-      return rpcManager.blocking(rsp);
-   }
-
-   private int clusterSize() {
-      return rpcManager.getTransport().getMembers().size();
+      CompletionStage<Void> rsp = rpcManager.invokeCommandOnAll(command, responseCollector, rpcOptions);
+      rpcManager.blocking(rsp);
    }
 
    @Override
@@ -369,40 +364,4 @@ public Collection<MetricInfo> getCustomMetrics(GlobalMetricsConfiguration config
    private interface Operation {
       void execute() throws Throwable;
    }
-
-   private record XSiteResponse(List<Address> errors) {
-   }
-
-   private static class VoidResponseCollector extends ValidResponseCollector<XSiteResponse> {
-
-      private final List<Address> errors;
-
-      private VoidResponseCollector(int expectedSize) {
-         errors = new ArrayList<>(expectedSize);
-      }
-
-      @Override
-      public XSiteResponse finish() {
-         return new XSiteResponse(errors);
-      }
-
-      @Override
-      protected XSiteResponse addValidResponse(Address sender, ValidResponse response) {
-         //no-op
-         return null;
-      }
-
-      @Override
-      protected final XSiteResponse addTargetNotFound(Address sender) {
-         //ignore leavers
-         return null;
-      }
-
-      @Override
-      protected final XSiteResponse addException(Address sender, Exception exception) {
-         errors.add(sender);
-         return null;
-      }
-   }
-
 }
```

I believe this should still have the same behavior. At least no tests complain about the change.

Reply:
The `Throwable` message will never include the failed node list. We probably aren't testing the error message anywhere.

Reply:
I see what you mean now :+1:. I thought those would be equivalent. Going to look at the other change and I think it's good.